fili-core/src/main/java/com/yahoo/bard/webservice/util/TableUtils.java

```diff
@@ -43,6 +46,21 @@
     }
 
     /**
+     * Get the schema column names from the dimensions and metrics.
+     *
+     * @param request  A request which supplies grouping dimensions and filtering dimensions
+     * @param query  A query model which has metric column and possibly dimension column names
+     *
+     * @return a set of strings representing schema column names
+     */
+    public static Set<String> getColumnNames(DataApiRequest request, DruidAggregationQuery<?> query) {
+        return Stream.of(
+                getDimensions(request, query).map(Dimension::getApiName),
+                query.getDependentFieldNames().stream()
+        ).flatMap(Function.identity()).collect(Collectors.toSet());
```

Comment:
This can be simplified using the `Stream.concat` function:

```java
return Stream.concat(
         getDimensions(request, query).map(Dimension::getApiName),
         query.getDependentFieldNames().stream
)
         .collect(Collectors.toSet());
```